source/extensions/transport_sockets/alts/tsi_socket.cc

```diff
@@ -361,7 +362,9 @@ void TsiSocket::closeSocket(Network::ConnectionEvent) {
 
 void TsiSocket::onConnected() {
   ASSERT(!handshake_complete_);
-  doHandshakeNext();
+  if (!downstream_) {
+    doHandshakeNext();
+  }
```

Comment:
Can you add some comments on why we defer this now? It's not immediately clear to me why we need this change here but not other transport sockets? Thank you.

/wait

Reply:
This was never called on the downstream. Other sockets don't do work here AFAICT.

Reply:
I see, OK.